pkg/upgrade/update.go

```diff
@@ -12,49 +12,83 @@ import (
 	"github.com/openshift/openshift-azure/pkg/api"
 	"github.com/openshift/openshift-azure/pkg/log"
 	"github.com/openshift/openshift-azure/pkg/util/azureclient"
+	"github.com/openshift/openshift-azure/pkg/util/managedcluster"
 )
 
 func (u *simpleUpgrader) Update(ctx context.Context, cs *api.OpenShiftManagedCluster, azuretemplate map[string]interface{}, deployFn api.DeployFn) error {
 	err := u.createClients(ctx, cs)
 	if err != nil {
-		return err
+		return &api.PluginError{Err: err, Step: api.PluginStepClientCreation}
 	}
-	// Deploy() may change the number of VMs.  If we can see that any VMs are
+	// deployFn() may change the number of VMs.  If we can see that any VMs are
 	// about to be deleted, drain them first.  Record which VMs are visible now
-	// so that we can detect newly created VMs and wait for them to become
-	// ready.
+	// so that we can detect newly created VMs and wait for them to become ready.
+	vmsBefore, err := u.getNodesAndDrain(ctx, cs)
+	if err != nil {
+		return &api.PluginError{Err: err, Step: api.PluginStepDrain}
+	}
+	err = deployFn(ctx, azuretemplate)
+	if err != nil {
+		return &api.PluginError{Err: err, Step: api.PluginStepDeploy}
+	}
+	err = u.InitializeCluster(ctx, cs)
+	if err != nil {
+		return &api.PluginError{Err: err, Step: api.PluginStepInitialize}
+	}
+	err = managedcluster.WaitForHealthz(ctx, cs.Config.AdminKubeconfig)
```

Comment:
We probably need an issue to enable updates of unresponsive clusters. @jim-minter thoughts?